internal/controllers/gateway/route_utils.go

```diff
@@ -142,20 +143,38 @@ func getSupportedGatewayForRoute[T types.RouteT](ctx context.Context, mgrc clien
 					if !(listener.Protocol == HTTPProtocolType || listener.Protocol == HTTPSProtocolType) {
 						continue
 					}
+					// if the listener has TLS enabled, but the protocol is not HTTPS, it is malformed, thus we can skip it.
+					if listener.TLS != nil && listener.Protocol != HTTPSProtocolType {
+						continue
+					}
```

Comment:
Well, I would prefer to at least log an warning.

Reply:
The problem is that logging something (or even returning an error) here doesn't make much sense: having a listener incompatible (for any reason) with the current HTTPRoute is not an error and cannot be seen as it would be. So in this function, we iterate over all the gateways referenced by the HTTPRoute, and we gather all the ones matching compatibility criteria. If no listener matches the current Route, we return a generic error.